### PR TITLE
feat: support `pd.Series.explode` with `ExtensionArray._explode` method

### PR DESCRIPTION
**Pull request by douglasdavis**
Fix for #38 

Now that pandas 2.2.0 is out we can implement this

### CODE REVIEW

File: src/awkward_pandas/array.py [hunk 1]
```diff
@@ -108,6 +108,14 @@ def _binop(self, other):
     def _reduce(self, name: str, *, skipna: bool = True, axis=None, **kwargs):
         return getattr(ak, name)(self._data, **kwargs)
 
+    def _explode(self):
+        nums = ak.num(self._data, axis=1)
+        nums_filled = ak.fill_none(nums, 0)
+        data = ak.where(nums_filled == 0, [[None]], self._data)
+        flat = ak.flatten(data)
+        arr = type(self)(flat)
+        return arr, ak.num(data, axis=1)
```
Comment (martindurant) [inline src/awkward_pandas/array.py:117]: When done on a series, what's the return type of explode()? Just curious.

Comment (douglasdavis) [reply src/awkward_pandas/array.py:117]: Just a `Series` again. The two returns here are consumed by `Series.explode`, it uses the second return to build a new `Index` object and then passes the array and the index objects to `Series._constructor`:

https://github.com/pandas-dev/pandas/pull/54834/files#diff-a5257444a1b322d619680fc77361cc6ea11ef36b363b4bb2289fdef0f41feb70R4403

**Comment by CrfzdPQM6** [comment]
@martindurant suggested I paste this snippet here as a useful test:
```
import awkward as ak
values = [ak.Array([1.0,2.0,3.0]), ak.Array([4.0,5.0]), ak.Array([6.0])]
df = pd.DataFrame({'pt':values})
```
because the output of `df.dtypes` is that `pt` is `object`

I'm running into issues with pandas 2.2.1 that df.explode() does not work for a column with dtype `awkward`

**Comment by martindurant** [comment]
> df = pd.DataFrame({'pt':values})

needs to have dtype specified (or astype called) to make it awkward, else you are simply calling pandas' default iteration, which would also work for python lists.

**Pull request merged**